File: backend/src/main/java/com/sims/entity/TeacherCourse.java
package com.sims.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("teacher_course")
public class TeacherCourse {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long teacherId;
    private Long courseId;
    private String semester;
    private String academicYear;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: frontend/src/api/teacherCourse.ts
import http from './index'

export interface TeacherCourse {
  id?: number
  teacherId: number
  courseId: number
  semester: string
  academicYear: string
  createdAt?: string
  updatedAt?: string
}

export function myTeacherCourses() {
  return http.get('/teacher-courses/my').then(r => r.data)
}

export function listTeacherCourses(params?: { teacherId?: number; semester?: string }) {
  return http.get('/teacher-courses', { params }).then(r => r.data)
}

export function createTeacherCourse(data: TeacherCourse) {
  return http.post('/teacher-courses', data).then(r => r.data)
}

export function deleteTeacherCourse(id: number) {
  return http.delete(`/teacher-courses/${id}`).then(r => r.data)
}
